tests/test_h2_client.c

```diff
@@ -5336,6 +5336,59 @@ TEST_CASE(h2_client_manual_data_write) {
     return s_tester_clean_up();
 }
 
+TEST_CASE(h2_client_manual_data_write_not_enabled) {
+
+    ASSERT_SUCCESS(s_tester_init(allocator, ctx));
+
+    struct aws_http_message *request = aws_http2_message_new_request(allocator);
+    ASSERT_NOT_NULL(request);
+
+    struct aws_http_header request_headers_src[] = {
+        DEFINE_HEADER(":method", "GET"),
+        DEFINE_HEADER(":scheme", "https"),
+        DEFINE_HEADER(":path", "/"),
+    };
+    aws_http_message_add_header_array(request, request_headers_src, AWS_ARRAY_SIZE(request_headers_src));
+    struct aws_http_make_request_options request_options = {
+        .self_size = sizeof(request_options),
+        .request = request,
+        .http2_use_manual_data_writes = false,
+    };
+    struct aws_http_stream *stream = aws_http_connection_make_request(s_tester.connection, &request_options);
+    ASSERT_NOT_NULL(stream);
+
+    aws_http_stream_activate(stream);
+
+    struct aws_byte_buf payload;
+    aws_byte_buf_init(&payload, allocator, 1024);
+
+    struct h2_client_manual_data_write_ctx test_ctx = {
+        .allocator = allocator,
+        .data = payload,
+    };
+
+    /* Try writing the data */
+    struct aws_input_stream *data_stream = s_h2_client_manual_data_write_generate_data(&test_ctx);
+    int64_t stream_length = 0;
+    ASSERT_SUCCESS(aws_input_stream_get_length(data_stream, &stream_length));
+    struct aws_http2_stream_write_data_options write_options = {
+        .data = data_stream,
+    };
+    ASSERT_FAILS(aws_http2_stream_write_data(stream, &write_options));
+    ASSERT_INT_EQUALS(aws_last_error(), AWS_ERROR_HTTP_MANUAL_WRITE_NOT_ENABLED);
```

Comment:
we have a special assert for this
```suggestion
    ASSERT_ERROR(AWS_ERROR_HTTP_MANUAL_WRITE_NOT_ENABLED, aws_http2_stream_write_data(stream, &write_options));
```
or keep it as two lines, but in our macros you put the expected value first (if it fails it prints out that it expected the value to be A but got B)
```suggestion
    ASSERT_FAILS(aws_http2_stream_write_data(stream, &write_options));
    ASSERT_INT_EQUALS(AWS_ERROR_HTTP_MANUAL_WRITE_NOT_ENABLED, aws_last_error());
```